homework/index.js

```diff
@@ -30,14 +30,105 @@
+    repositoryDetails.forEach((repository, index) => {
+      createAndAppend('option', select, {
+        text: repository.name,
+        value: index,
+      });
+    });
+  }
+
+  function addRow(tbody, labelText, value, str) {
+    const tr = createAndAppend('tr', tbody);
+    createAndAppend('td', tr, { text: `${labelText}`, class: str });
+    createAndAppend('td', tr, { text: value });
+    return tr;
+  }
+
+  function renderRepoInformation(repositories, container) {
+    container.innerHTML = ' ';
+    const table = createAndAppend('table', container, { class: 'infoTable' });
+    const tbody = createAndAppend('tbody', table);
+    const firstRow = addRow(tbody, 'Repository: ', '', 'explanations');
+    createAndAppend('a', firstRow.lastChild, {
+      href: `${repositories.html_url}`,
+      target: 'blank',
+      text: `${repositories.name}`,
+    });
+    addRow(tbody, 'Description:', repositories.description, 'explanations');
+    addRow(tbody, 'Fork:', repositories.forks, 'explanations');
+    addRow(tbody, 'Updated:', `${repositories.updated_at}`.substring(0, 10), 'explanations');
+  }
+
+  function renderContributorsInformation(repositories, container) {
+    const contributors = repositories.contributors_url;
+    const url = contributors;
+    fetchJSON(url, (error, contributorDetails) => {
+      if (error) {
+        renderError(error);
+        return;
+      }
+
+      container.innerHTML = ' ';
+      const div = document.getElementById('cont-ulist');
+      createAndAppend('h4', div, { text: 'Contributors' });
+      const contrList = createAndAppend('ul', div, { class: 'contr-list' });
+      contributorDetails.forEach(contributor => {
+        const li = createAndAppend('li', contrList);
+        const a = createAndAppend('a', li, { href: contributor.html_url, target: 'blank' });
+        createAndAppend('img', a, {
+          src: contributor.avatar_url,
+          class: 'images',
+        });
+        createAndAppend('span', a, {
+          text: contributor.login,
+        });
+        createAndAppend('span', a, {
+          text: contributor.contributions,
+          class: 'contributions',
+        });
+      });
+    });
+  }
+
   function main(url) {
+    const root = document.getElementById('root');
+    const menuSection = createAndAppend('section', root, { id: 'menu-section' });
+    createAndAppend('p', menuSection, { text: 'HYF Repositories', id: 'hyf-text' });
+    const select = createAndAppend('select', menuSection, { id: 'select-button' });
+    const bodyDiv = createAndAppend('div', root, { id: 'body-div' });
+    const section = createAndAppend('section', bodyDiv, { class: 'repo-info-list' });
+    const contrDiv = createAndAppend('div', bodyDiv, {
+      class: 'contributors-div',
+      id: 'cont-ulist',
+    });
+
     fetchJSON(url, (err, repositories) => {
-      const root = document.getElementById('root');
+      repositories.sort((a, b) => a.name.localeCompare(b.name));
```

Comment:
If there is an error this `sort` will produce a runtime error in the console, as there is no data to sort. You better wait with sorting until you have established that there was no error.